amethyst_network/src/simulation/transport/laminar.rs

```diff
@@ -0,0 +1,186 @@
+            LaminarNetworkRecvSystem,
+            NETWORK_RECV_SYSTEM_NAME,
+            &[NETWORK_POLL_SYSTEM_NAME],
+        );
+        builder.add(
+            NetworkSimulationTimeSystem,
+            NETWORK_SIM_TIME_SYSTEM_NAME,
+            &[NETWORK_RECV_SYSTEM_NAME],
+        );
+        world.insert(LaminarSocketResource::new(self.socket));
+        Ok(())
+    }
+}
+
+struct LaminarNetworkSendSystem;
+
+impl<'s> System<'s> for LaminarNetworkSendSystem {
+    type SystemData = (
+        Write<'s, TransportResource>,
+        Write<'s, LaminarSocketResource>,
+        Read<'s, NetworkSimulationTime>,
+    );
+
+    fn run(&mut self, (mut transport, mut socket, sim_time): Self::SystemData) {
+        socket.get_mut().map(|socket| {
+            let messages = transport.drain_messages_to_send(|_| sim_time.should_send_message_now());
+
+            for message in messages.iter() {
+                let packet = match message.delivery {
+                    DeliveryRequirement::Unreliable => {
+                        Packet::unreliable(message.destination, message.payload.to_vec())
+                    }
+                    DeliveryRequirement::UnreliableSequenced(stream_id) => {
+                        Packet::unreliable_sequenced(
+                            message.destination,
+                            message.payload.to_vec(),
+                            stream_id,
+                        )
+                    }
+                    DeliveryRequirement::Reliable => {
+                        Packet::reliable_unordered(message.destination, message.payload.to_vec())
+                    }
+                    DeliveryRequirement::ReliableSequenced(stream_id) => {
+                        Packet::reliable_sequenced(
+                            message.destination,
+                            message.payload.to_vec(),
+                            stream_id,
+                        )
+                    }
+                    DeliveryRequirement::ReliableOrdered(stream_id) => Packet::reliable_ordered(
+                        message.destination,
+                        message.payload.to_vec(),
+                        stream_id,
+                    ),
+                    DeliveryRequirement::Default => Packet::reliable_ordered(
+                        message.destination,
+                        message.payload.to_vec(),
+                        None,
+                    ),
+                };
+
+                if let Err(e) = socket.send(packet) {
+                    error!("There was an error when attempting to send packet: {:?}", e);
+                }
+            }
+        });
+    }
+}
+
+struct LaminarNetworkPollSystem;
+
+impl<'s> System<'s> for LaminarNetworkPollSystem {
+    type SystemData = Write<'s, LaminarSocketResource>;
+
+    fn run(&mut self, mut socket: Self::SystemData) {
+        socket
+            .get_mut()
+            .map(|socket| socket.manual_poll(Instant::now()));
```

Comment:
So I am wondering whether we should do something with time zones. Like maybe want to poll based on UTC. Because we are having multiple systems running over the globe.

Reply:
Eh. That can be future work.